### ISSUE TITLE
Modifying worn clothing doesn't update player info.

### ISSUE THREAD

**Issue by chrispikula** [CleverRaven/Cataclysm-DDA#54303]
### Describe the bug

Modifying clothing (as it currently stands, v. ), doesn't update the Player Info screen if the encumbrance changes.  
One needs to remove the modified clothing, and then put it back on again to update Player Info.

### Steps to reproduce

Modify a worn piece of clothing that only requires one item to modify.  (Due to #53819)
See preview that encumbrance/warmth increases.
Check Player Info.  No change.
Take off, put back on item. 
Check Player Info.  Changed.

### Expected behavior

Probably shouldn't be modifying clothing without taking it off and putting it back on.  Especially skin layer stuff.

### Versions and configuration

Win 10,
Tiles,
v. a34653b
Base game.

### Additional context

Probably should be fixed in the same pass as #53819.

**Comment by LeahLuong** [comment]
I don't think this is the only issue here. See #50915. Not sure who's best to @ here . . .

**Comment by bombasticSlacks** [comment]
> I don't think this is the only issue here. See #50915. Not sure who's best to @ here . . .

I believe this is fixed.

Encumbrance is a cached value since it changes as the volume of pockets changes. The modify code probably just doesn't invalidate encumbrance like other parts do. I would guess it just needs to update character::set_check_encumbrance as true so that the next time the UI is opened the item recalculates.
